### ISSUE TITLE
[PATCH] Add old style RTs and Reply-to-all [also some other fixes]

### ISSUE THREAD

**Issue by GoogleCodeExporter** [marig345/prpltwtr#36]
```
This patch will do the following:

* Add old style RTs and Reply-to-all options
* Abstract icon size [and change it to 32]
* Fix the crash when replying if disconnected
* Improve rate limit efficiency (request more tweets per request)
* Remove all compilation warnings

It applies cleanly to 0.5.2. I tested it with mingw32, compiling on Linux for 
Windows. I've been running it for a few days with no problems.

Also attached are DLLs for Windows
```

Original issue reported on code.google.com by `mikeage` on 5 Dec 2010 at 6:42

Attachments:
- [prpltwtr-0.5.2.diff](https://storage.googleapis.com/google-code-attachments/prpltwtr/issue-36/comment-0/prpltwtr-0.5.2.diff)
- [gtkprpltwtr.dll](https://storage.googleapis.com/google-code-attachments/prpltwtr/issue-36/comment-0/gtkprpltwtr.dll)
- [prpltwtr.dll](https://storage.googleapis.com/google-code-attachments/prpltwtr/issue-36/comment-0/prpltwtr.dll)

**Comment by GoogleCodeExporter** [comment]
```
so does this need to be compiled or will these files do the trick?
```

Original comment by `TheMexicanTaint` on 11 Dec 2010 at 6:29

**Comment by GoogleCodeExporter** [comment]
```
If you want to compile, be my guest. However, all the changes are included in 
the compiled DLLs.
```

Original comment by `mikeage` on 11 Dec 2010 at 6:43

**Comment by GoogleCodeExporter** [comment]
```
when I attempt to use the old retweet function, pidgin crashes. any ideas?
```

Original comment by `TheMexicanTaint` on 23 Dec 2010 at 1:12

**Comment by GoogleCodeExporter** [comment]
```
sorry here's more info I forgot to add to help fix the problem:

OS: Win 7 32 bit
version: Pidgin 2.7.8 (libpurple 2.7.8)
prpltwtr version: 0.5.2
```

Original comment by `TheMexicanTaint` on 23 Dec 2010 at 1:47

**Comment by GoogleCodeExporter** [comment]
```
Are you connected to twitter at the time?

Can you run pidgin in debug mode (add --debug to the command line), and send me 
the last few lines (maybe 15 - 20 or so)?

Thanks
```

Original comment by `mikeage` on 23 Dec 2010 at 1:50

**Comment by GoogleCodeExporter** [comment]
```
yes I was connected to twitter, on the home timeline and attempted and old 
style retweet. 

Can I alternatively just open the debug window and save it? I've never actually 
ran pidgin in command line mode before.
```

Original comment by `TheMexicanTaint` on 23 Dec 2010 at 2:05

**Comment by GoogleCodeExporter** [comment]
```
The problem with the debug window is that once the crash occurs, you won't be 
able to save it. 

Right click on the shortcut, and add --debug
```

Original comment by `mikeage` on 23 Dec 2010 at 2:07

**Comment by GoogleCodeExporter** [comment]
```
On second thought I didn't have the home timeline open when this error occured. 
Does it need to be in order for it to work? I tried the old retweet from a 
separate window. 

(Just trying to save you work in case it needs to be open for old retweets to 
work)
```

Original comment by `TheMexicanTaint` on 23 Dec 2010 at 2:29

**Comment by GoogleCodeExporter** [comment]
```
Very possible. There's an issue with not having the "right" chat window open 
that I've noticed; not sure what the solution is yet. I've seen it most when 
replying / retweeting from a "conversation" window. 
```

Original comment by `mikeage` on 23 Dec 2010 at 3:10

**Comment by GoogleCodeExporter** [comment]
```
did the debug in command line, replicated the issue and it all crashed before i 
could save... 
```

Original comment by `TheMexicanTaint` on 23 Dec 2010 at 3:10

**Comment by GoogleCodeExporter** [comment]
```
Ok, I just reproduced it. It happens whenever I try to "old retweet" from a 
mentions window. I will try to fix it.
```

Original comment by `mikeage` on 28 Dec 2010 at 3:16

**Comment by GoogleCodeExporter** [comment]
```
thanks mikeage, appreciate it
```

Original comment by `TheMexicanTaint` on 28 Dec 2010 at 3:29

**Comment by GoogleCodeExporter** [comment]
```
Here's two new DLLs [although only gtkprpltwtr.dll should have changed] and a 
new .diff. It was actually really easy to ID the bug; fixing it required 
learning a bit about prpltwtr and pidgin in general, but it wasn't bad.

let me know if you still have any problems. There's one more crash hiding here 
that seems to relate to sleep modes that I want to tackle next...
```

Original comment by `mikeage` on 28 Dec 2010 at 5:22

Attachments:
- [prpltwtr-0.5.2_fix.diff](https://storage.googleapis.com/google-code-attachments/prpltwtr/issue-36/comment-13/prpltwtr-0.5.2_fix.diff)
- [prpltwtr.dll](https://storage.googleapis.com/google-code-attachments/prpltwtr/issue-36/comment-13/prpltwtr.dll)
- [gtkprpltwtr.dll](https://storage.googleapis.com/google-code-attachments/prpltwtr/issue-36/comment-13/gtkprpltwtr.dll)

**Comment by GoogleCodeExporter** [comment]
```
it works... by opening up the home timeline and sending it through there. its 
not an optimal choice but then again its better than nothing at all, good work. 
thanks again.
```

Original comment by `TheMexicanTaint` on 28 Dec 2010 at 9:42

**Comment by GoogleCodeExporter** [comment]
```
I chose to do it that way since old retweets are just regular messages, and not 
part of the conversation [they're also not "in-reply-to"]. It's both logically 
consistent  and much easier [since anything sent in a private IM window is sent 
as a reply in prpltwtr].

If you look at other clients, you can confirm the way old retweets work; 
they're just a new tweet that happens to mention the original person, but not a 
reply.
```

Original comment by `mikeage` on 29 Dec 2010 at 4:10

**Comment by GoogleCodeExporter** [comment]
```
Just realized there was a small typo in the patch; line 165 should read:

if (memcmp(&text[i+1], username+1, j-(i+1)) && memcmp(&text[i+1], user, 
j-(i+1))) {

Note the +1 after username.

Now reply-to-all no longer includes you.

Also a new DLL attached 
```

Original comment by `mikeage` on 10 Jan 2011 at 3:38

Attachments:
- [gtkprpltwtr.dll](https://storage.googleapis.com/google-code-attachments/prpltwtr/issue-36/comment-16/gtkprpltwtr.dll)

**Comment by GoogleCodeExporter** [comment]
```
Issue 17 has been merged into this issue.
```

Original comment by `mikeage` on 14 Feb 2011 at 6:38

**Comment by GoogleCodeExporter** [comment]
```
Issue 25 has been merged into this issue.
```

Original comment by `mikeage` on 14 Feb 2011 at 6:43

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 0.6.0
```

Original comment by `mikeage` on 14 Feb 2011 at 6:46
- Changed state: **Verified**
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**
